redhat-appstudio/service-provider-integration-operator
# Svpi137 vault auth

### What does this PR do?
Updates Vault tokenstorage configuration
 - ready for approle authentication method
 - config options moved to args/env variables, removed from config file
 - updated k8s and minikube deployment to add needed env to containers
 - use approle vault auth on kubernetes/minikube

deployment patches will dissappear when we switch to approle everywhere. But it's not ready for openshift and infra-deployment yet.


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-137

### How to test this PR?


operator image `quay.io/mvala/spi-operator:svpi137-vaultAuth`
oauth image `quay.io/mvala/spi-oauth:svp137-vaultAuth`

1. deploy on minikube
2. check 
  - deployment has env variables configured with vault host and approle vault auth
  - 2 secrets with vault approle creds are created in spi-system namespace
  - configuration secret has no vault options
  - both deployments starts fine and connects to Vault via Ingress
